### ISSUE TITLE
AriaAnnouncer not working in my library

### ISSUE THREAD

**Issue by oravecz** [jpmorganchase/salt-ds#4627]
### Latest version

- [x] I have tested the latest version

### Description

Using the AriaAnnouncerProvider and useAriaAnnouncer() hook in my library seems straight forward. I have ensured that my message is being sent to the context, and I modified the styles of the aria-live div element so I can see my message appear. I even tried increasing the `ARIA_ANNOUNCE_DELAY` so my message is visible for longer.

No matter what I have tried, my message is not spoken by the screen reader. 

### Steps to reproduce

This is an internal JPMC codebase, so if a developer would like to pair while I demonstrate - that would be fine.

### Package name(s)

Core (@salt-ds/core)

### Package version(s)

1.39.0

### Browser

- [x] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Microsoft Edge

### Operating system

- [x] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### Are you a JPMorgan Chase & Co. employee?

- [x] I am an employee of JPMorgan Chase & Co.

**Comment by joshwooding** [comment]
Could you share if you are using the latest version of MacOS and have you only tried in Chrome?

**Comment by mark-tate** [comment]
can you provide a minimal reproduction of the issue ?

**Comment by mark-tate** [comment]
https://jpmc.stackenterprise.co/questions/100110

**Comment by chrispcode** [comment]
I narrowed the issue to:
1. The `AriaAnnouncerProvider` needs to be present on a parent component if you want to use the `useAriaAnnoucer` hook. Check the example here: https://github.com/jpmorganchase/salt-ds/blob/main/packages/core/stories/aria-announcer/aria-announcer.stories.tsx

2. The `AriaAnnouncerContext.Consumer ` can be used instead in the same component, but keep in mind that there is additional functionality in the hook (mainly debouncing) that will not reflect in the object you get from the consumer. 

```tsx
function Example() {
  function addItem(announcer: AriaAnnouncer | undefined) {
    // run some logic

    announcer?.announce("An Item was added")
  }

  return (
    <AriaAnnouncerProvider>
      <AriaAnnouncerContext.Consumer>
        {(announcer) => (
          <Button onClick={() => addItem(announcer)}>Add Item</Button>
        )}
        </AriaAnnouncerContext.Consumer>
      </AriaAnnouncerProvider>
  )
}
```

**Comment by chaithalbe** [comment]
We do have the AriaAnnouncerProvider on the [parent here](https://bitbucketdc-cluster07.jpmchase.net/projects/PEPPER/repos/pepper-dashboard/commits/d2bf1a07db23b9eba386b14c9a2a97dfae4ad53a#src/dashboard/Dashboard.tsx). I tried using the Consumer as well, but it doesn't work. 
I also tried moving the Provider directly in the Gallery component, but it works intermittently, not every time an item is added.

**Comment by joshwooding** [comment]
I've been able to get this working on Storybook, if there's an issue when distributing the library it will likely be due to multiple versions of @salt-ds/core in the application or that @salt-ds/core is being bundled into the library when it should be treated as external.
